lib/number.js

```diff
@@ -0,0 +1,318 @@
+import {
+	getDocumentLocaleSettings,
+	getLanguage,
+	merge,
+	validateFormatValue
+} from './common.js';
+
+function formatPositiveInteger(value, descriptor) {
+	value = Math.floor(value);
+
+	const valueStr = '' + value;
+	let ret = '';
+
+	const groupSizes = [3];
```

Comment:
Every locale uses a group size of `3`, so I simplified the code here a bit. I may go through and simplify it further at some point.